### ISSUE TITLE
What is the best way to capture backup traffic?

### ISSUE THREAD

**Issue by Yaldo425** [horrorho/InflatableDonkey#63]
A long time ago people would capture the traffic by installing a certificate and jailbreaking the phone to backup the keychain containing the certificate. Afterwards, the phone would be restored, re-jailbroken, and the keychain restored so that the backup processes traffic could be captured. 

More recently, I read about creating a profile with Apple Configurator and being able to add it during the setup assistant provided that HTTPS decryption is turned off during the login part and back on before accepting the EULA, but my problem is it says that there are no backups available when I'm capturing traffic, so I am not able to restore.

How is it done right now? Is there anything different being done now with the second method?

**Comment by Louti** [comment]
You can do so by jailbreaking your device and installing KillSwitchv2 so it bypasses the SSL pinning

**Comment by Yaldo425** [comment]
@Louti 

After installing SSLKillSwitchv2 and using a method to get back to the setup screen, I still cannot get a backup to show. I am presented with the following error:

```
Cannot Load Backups
There was a problem loading your iCloud backups. Try again, set up as a new iPad, or restore from an iTunes backup.
```
What I'm really asking is how is it possible to get all of the restore protocols and APIs for IOS 10 when there isn't even a jailbreak for?

**Comment by horrorho** [comment]
@Yaldo425 Hi. I was given iOS 9 capture traffic way back. I've never seen an iOS 10 restore log.

> What I'm really asking is how is it possible to get all of the restore protocols and APIs for IOS 10 when there isn't even a jailbreak for it?

The process is largely similar to iOS 9. InflatableDonkey also logs heavily and reports on things which it doesn't understand, this combined with reversing iOS binaries is how I've kept this tool functional.

**Comment by Jirmi** [comment]
Hi @Yaldo425, just wondering whether you have been successful in capturing the traffic on the jailbroken device while applying sslkillswitch?

**Comment by Yaldo425** [comment]
@Jirmi Yes, I am able to capture the traffic with sslkillswitch. After this issue was created I found out that I can directly capture backup traffic by deleting com.apple.purplebuddy in IOS, which brings me back to the set up screen where I can choose to restore the device from a backup and capture all the logs.

**Comment by Jirmi** [comment]
Thanks @Yaldo425! In my case despite deleting com.apple.purplebuddy.plist the Setup app doesn't display the screen with options for restoration (icloud, itunes or setup as new device). Could you please tell me how did you make it work please?

**Comment by Yaldo425** [comment]
@Jirmi What version of iOS are you?

In order to get it to work, you'll need to do the following:

- Go to /var/mobile/Library/Preferences and delete com.apple.PurpleBuddy.notbackedup.plist and com.apple.PurpleBuddy.plist

-Go to /var/mobile/Library/Caches and delete the com.apple.purplebudy folder.

After that just reboot the device


A problem I am currently struggling with is getting it to work on the Phoenix 9.3.5 jailbreak. The setup.app won't show up without a reboot and the jailbreak is semi-tethered.

**Comment by Jirmi** [comment]
I've done these steps but without rebooting device since I'm testing it on iOS 10.2 (jailbroken with Yalu102) which is tethered tool. I believe if I reboot the device I will lose the functionality of ioskillswitch. Or am I wrong? Actually, I though you were successful on iOS 10 already.

**Comment by Yaldo425** [comment]
@Jirmi No, you are right, you would lose the functionality. When I was doing my testing I was running on iOS 8 with an untethered jailbreak. I'm trying my best to figure something out because I am in need of this as well.

**Comment by Jirmi** [comment]
Ah I see now :-) it all makes finally sense. Thanks for the clarification. I will let you know when I progress anyhow with the 10.2 since it could applicable also in your scenario.

**Comment by Yaldo425** [comment]
@Jirmi No problem! I asked around and was given a possible method, I couldn't get it to work myself but maybe it will work for you.

Get the Filza file manager application and head over to /Applications/Setup.app. Open Info.plist and delete `SBAppTags` and it's values. Once that's done, execute `uicache` via terminal on the device and respring with `killall SpringBoard` The Setup app should be on your homescreen after that.

**Comment by Jirmi** [comment]
Just gave it try but it doesn't seems to work. But thanks anyway, it was worth of shot ;-)

**Comment by Jirmi** [comment]
Managed to get the logs in the end but it takes while to set up the device properly.

**Comment by Yaldo425** [comment]
@Jirmi Was it with the decide you previously mentioned on the Yalu jailbreak? If so how did you end up getting it work? I had still attempted to find a workaround but was unsuccessful

**Comment by Jirmi** [comment]
Yes, it was with the device jailbroken with Yalu102. It was a little bit tricky and I'm not sure if I would be able to repeat the process properly again since it was a lots of tries and fails. Basically, I jailbroke the device, installed all tools via Cydia (killswitch), deleted the PurpleBuddy files (as you advised, also the caches). I also tried to delete the SBAppTags which only made the Setup app visible on springboard. Then I rebooted the device and re-jailbroke the device. Lastly, I set up the Charles proxy.
